src/main/groovy/wooga/gradle/version/internal/ToStringProvider.groovy

```diff
@@ -23,24 +23,25 @@ import org.gradle.api.provider.Provider
 
 class ToStringProvider <T> extends AbstractMinimalProvider<T> {
 
-    private final Provider<T> inner;
+    private final AbstractMinimalProvider<T> inner;
 
     ToStringProvider(Provider<T> provider) {
-        this.inner = provider
+        this.inner = provider as AbstractMinimalProvider<T>
```

Comment:
Is every implementation of `Provider` a `AbstractMinimalProvider` ? The reason why I used the public interface was to prevent potential breaks when Gradle changes. I know the irony that I myself extend from an internal class. But I don't expect that all wrapped providers are of the same base. Is there a way to make this code safer? I don't know from the top of my head which overrides are from the Provider interface and which came from the `AbstractMinimalProvider` class. To be honest I'm not even sure anymore if we could remove this provider all together and solve it differently.

Reply:
> Is every implementation of Provider a AbstractMinimalProvider ?

As far as I checked out, yes.

>  Is there a way to make this code safer? I don't know from the top of my head which overrides are from the Provider interface and which came from the AbstractMinimalProvider class

I remember running in the same issue when I was implementing some helper providers in gradle-commons, and there was something about using another provider `.get()` inside of `calculateOwnValue` that could go wrong, I just don't remember exactly what now, so in the end of the day, using the 'parent' function in this composition seems to me to be the safer way.

> To be honest I'm not even sure anymore if we could remove this provider all together and solve it differently.
I agree 100% with that, but this is being used in a sensitive part of the code (generating version number) and I didn't want to touch it very much, so...

Reply:
But why did you have to change it if you are not 100% sure it is needed? I mean it worked like this for the better part of 2 years.

Reply:
Fair enough, I'll rollback the old version.